src/parser.h

```diff
@@ -0,0 +1,147 @@
+#pragma once
+
+#include "datatypes.h"
+
+enum { MAX_YAML_LINE_LENGTH = 64 };
+enum { MAX_YAML_NUM_LINES = 1024 };
+enum { NUM_YAML_FILE_SECTIONS = 3 };  // rules, image location, tiles
+enum {
+  NUM_RULES = 2
+};  // allow 90 degree rotations, allow 180 degree
+    // rotations
+
+/**
+ * Get the text from the input yaml file
+ *
+ * Given the pointer to the user submitted yaml file, generate an array of
+ * arrays of strings containing the contents of the yaml file (as plaintext).
+ * The zeroth array of strings contains the "rules" section, the first the
+ * "image location" section, the second the "tiles" section.
+ *
+ * @param input_yaml_file The pointer to the file object representing the input
+ * yaml
+ *
+ * @return The array of arrays of strings containing the contents of the file
+ * split into sections
+ */
+char*** get_text_split_sections(FILE* input_yaml_file);
```

Comment:
Not convinced this needs to be a char***.
A new struct may be appropriate to handle discrete sections of the yaml.